framework/auth/views.py

```diff
@@ -155,7 +178,7 @@ def forgot_password_get(auth):
     return context
 
 
-def forgot_password_post():
+def forgot_password_post(mail_template=mails.FORGOT_PASSWORD, reset_route='reset_password_get'):
```

Comment:
I see these arguments are defined for both views `forgot_password_post` and `forgot_password_institution_post`. However, they always use the default values without taking advantage of this design. We can combine `forgot_password_post` and `forgot_password_institution_post` since the only differences are are email template and notification message. This not only reduces duplicate code but also makes future maintenance easier.

```python
def forgot_password_post(...)
    return _forgot_password_post(...)

def forgot_password_institution_post(...)
    return _forgot_password_post(...)

def _forgot_password_post(...)
    # everything goes here
```

Reply:
There is another *small* difference: the non-inst version puts the return message in an `alert`, while the inst version displays it as page block text.  That should be solvable with a new kwarg.  ISTR there was another reason I decided to c-n-p it, rather than reuse a single unified implementation, but I can't for the life of me remember what is was.  I will try reunifying again.  Thank you!

Reply:
Sounds good! I am also fine to leave them as they are if they can't be merged easily / without adding complexity.